critwall/src/main/java/org/opencadc/critwall/FileSync.java

```diff
@@ -125,20 +129,34 @@ public FileSync(Map<String,Object> daoConfig, ConnectionConfig connectionConfig,
         this.selector = selector;
         this.nthreads = nthreads;
 
+        // Notes on queue and thread pool instances:
+        // - LinkedBlockingQueue (jobQueue) is the queue used in this producer/consumer implementation.
+        // - jobQueue will block on put() if queue capacity is reached, and on take() if queue is
+        //  empty (useful at thread pool startup)
+        // - FileSync (this file) is the producer that adds to jobQueue (using BlockingQueueThreadPool.addTask())
+        // - BlockingQueueThreadPool's worker threads are the consumers (nthreads total)
+
+        // Justification for not using ExecutorService in this case is it's too complex, and
+        // the default implementations don't provide the behaviour needed. (it uses 'offer' instead
+        // of 'put' under the covers.)
+
```

Comment:
given comments in the ThreadPool class, some of the above will change

Reply:
changed